packages/core/src/geometry/GeometrySystem.js

```diff
@@ -586,22 +586,48 @@ export default class GeometrySystem extends System
      */
     draw(type, size, start, instanceCount)
     {
-        const { gl } = this;
+        const { gl, renderer: { context } } = this;
         const geometry = this._activeGeometry;
 
         // TODO.. this should not change so maybe cache the function?
 
         if (geometry.indexBuffer)
         {
-            if (geometry.instanced)
+            const byteSize = geometry.indexBuffer.data.BYTES_PER_ELEMENT;
+
```

Comment:
could add:
```const glType = byteSize === 2 ? gl.UNSIGNED_SHORT : gl.UNSIGNED_INT```

then we don't need the extra  if block

Reply:
updated